controllers/usersController.js

```diff
@@ -1,21 +1,36 @@
-const express = require('express');
-const { getAllUsers, getUserById } = require('../queries/usersQueries');
+const express = require("express");
+const { getAllUsers, getUserById } = require("../queries/usersQueries");
 
 const usersController = express.Router();
 
-usersController.get('/', (req, res) => {
-  users = getAllUsers() || [];
-  res.status(200).json({ data: users });
+usersController.get("/", async (req, res) => {
+  try {
+    users = (await getAllUsers()) || [];
+    if (users.length > 0) {
+      res.status(200).json({ data: users });
+      console.log("=== GET users: ", users, "===");
+    } else {
+      res.status(404).json({ error: `Users ${users} not found` });
```

Comment:
I appreciate the care and attention to detail here. You are handling the case where there are no users and we get an empty array back from the query. Actually, I think this is fine and I wouldn't call this a 404. You can just return a 200 with the empty array as the data. 

For sure it would be weird if an app had only one table and that table had nothing in it. But there are other cases where we would want to return an empty array and it wouldn't be weird.

For example, in our class app we have the endpoint `/student/:id/grades` which returns all of the grades for a particular student. It is possible that a student exists but they don't have any grades yet. In this case the right thing to do is return a 200 with `[]` as the data.

In general we should return a 404 only if:
- the client is trying to use a route + method that doesn't exist -or-
- the client is looking for a specific thing that doesn't exist (e.g., a student with id that isn't in our table)